### PR TITLE
Caddisfly dropdown remains in changed question testing(connect #1809)

### PR DESCRIPTION
**Pull request by bogere**
#### Before the PR (what is the issue or what needed to be done)
The Caddisfly test dropdown will remain in the screen even when the user selects another question type like free text , options when editing the question.
#### The solution
Caddisfly test dropdown does not show when you select another question type.
## Checklist
* [x] Connect the issue
* [x] Test plan
* [x] Copyright header
* [x] Code formatting
* [x] Documentation

### CODE REVIEW

File: Dashboard/app/js/lib/controllers/survey-controllers.js [hunk 1]
```diff
@@ -429,7 +429,8 @@ FLOW.projectControl = Ember.ArrayController.create({
    */
   loadCaddisflyResources: function () {
       var caddisflyResourceController = FLOW.router.get('caddisflyResourceController');
-      if (Ember.empty(caddisflyResourceController.get('content'))) {
+        if(Ember.empty(caddisflyResourceController.get('content'))){
+      //if (!Ember.empty(caddisflyResourceController.get('content'))) { //important for loading cadisfly
```
Comment (muloem) [inline Dashboard/app/js/lib/controllers/survey-controllers.js]: This comment is still here but doesn't need to be...

File: Dashboard/app/js/lib/views/surveys/question-view.js [hunk 2]
```diff
@@ -115,7 +115,7 @@ FLOW.QuestionView = FLOW.View.extend({
 
   amNoOptionsType: function () {
     var val;
-    if (!Ember.none(this.type)) {
+    if (!Ember.none(this.type)) {
```
Comment (muloem) [inline Dashboard/app/js/lib/views/surveys/question-view.js]: Please remove the whitespace at the end of the line. 👍

File: Dashboard/app/js/lib/views/surveys/question-view.js [hunk 3]
```diff
@@ -211,15 +210,17 @@ FLOW.QuestionView = FLOW.View.extend({
     }
 
     // reset selected caddisfly resource
-    FLOW.selectedControl.set('selectedCaddisflyResource', null);
+    FLOW.selectedControl.set('selectedCaddisflyResource', null);//do u need this like above
```
Comment (muloem) [inline Dashboard/app/js/lib/views/surveys/question-view.js]: Also remove these comments

File: Dashboard/app/js/lib/views/surveys/question-view.js [hunk 4]
```diff
@@ -211,15 +210,17 @@ FLOW.QuestionView = FLOW.View.extend({
...
     // if the caddisflyResourceUuid is not null, get the resource
     if (!Ember.empty(FLOW.selectedControl.selectedQuestion.get('caddisflyResourceUuid'))) {
       var caddResource = FLOW.router.caddisflyResourceController.content.findProperty('uuid', FLOW.selectedControl.selectedQuestion.get('caddisflyResourceUuid'));
       if (!Ember.empty(caddResource)) {
-        FLOW.selectedControl.set('selectedCaddisflyResource',caddResource);
+        FLOW.selectedControl.set('selectedCaddisflyResource',caddResource);// u selected it here thus
```
Comment (muloem) [inline Dashboard/app/js/lib/views/surveys/question-view.js]: comments and console log not needed.

File: Dashboard/app/js/lib/views/surveys/question-view.js [hunk 5]
```diff
@@ -860,7 +861,7 @@ FLOW.QuestionView = FLOW.View.extend({
     this.set('showAddAttributeDialogBool', false);
   },
 
-  validateQuestionObserver: function(){
+  validateQuestionObserver: function(){ //this code makes questionValidationFailure true after observing text
```
Comment (muloem) [inline Dashboard/app/js/lib/views/surveys/question-view.js]: comment. 🔫

File: Dashboard/app/js/lib/views/surveys/question-view.js (hunk 2 above)
Comment (bogere) [reply Dashboard/app/js/lib/views/surveys/question-view.js]: iam not sure about the end of the line you are referring too.

**Pull request merged**